brave/brave-core
# Add button/informer to switch Android system sync toggle

Android system settings have an option to enable/disable sync for whole system. When that flag is turned off, Brave sync does not work on Android. 

We have a QA article https://support.brave.com/hc/en-us/articles/360047642371-Sync-FAQ `Why won’t my data Sync on my Android device?` , but users need more clear way to understand what is happening. Chrome informs about such case in browser preferences. We may want also to have that.

This PR adds an informer which can be pressed and leads to system settings.

The screenshots are below.


Resolves https://github.com/brave/brave-browser/issues/11128

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.

## Test Plan:

I. Enabling sync when system auto-sync is turned off

1. Turn off Android system auto sync data switch
2. Launch Brave 
3. Start new sync chain, choose `Add computer`, `Done`
4. You must see the informer and one device in the list
5. Press informer, ensure you are in system settings
6. Switch the toggle `Auto sync data` on
7. Press back
8. Ensure you are in Brave sync settings and now there are no informer
9. Ensure you can sync bookmarks by adding 2nd device into chain

II. Sync chain was established, but Android system settings auto sync data is turned off

1. Make sync chain, one of devices is Android
2. On Android go to system settings and toggle off switch `auto sync data`
3. On Brave browser go to sync settings, ensure informer is shown 
4. Press informer, ensure you are in system settings
5. Switch the toggle `Auto sync data` on
6. Press back
7. Ensure you are in Brave sync settings and now there are no informer

III. Popup informer
1. Make sync chain, one of devices is Android
2. On Android go to system settings and toggle off switch auto sync data
3. Swipe out Brave from recent apps
4. Launch Brave again - ensure the popup informer is shown 
5. Press OK
6. Relaunch Brave - ensure popup informer is shown 
7. Press Open Settings - ensure you've been brought to Android system Accounts sync settings 
8. Relaunch Brave - ensure popup informer is shown 
9. Press Don't show again
10. Relaunch Brave - ensure popup informer is not shown
